prediction_market_agent/agents/microchain_agent/memory.py

```diff
@@ -99,3 +95,78 @@ def search(
             from_=from_,
             to=to,
         )
+
+
+class ChatHistory(BaseModel):
+    """
+    A collection of chat messages. Assumes that the chat messages are ordered by
+    datetime.
+    """
+
+    chat_messages: Sequence[ChatMessage]
+
+    def add_message(self, chat_message: ChatMessage) -> None:
+        list(self.chat_messages).append(chat_message)
+
+    @staticmethod
+    def from_list_of_dicts(list_of_dicts: list[Dict[str, str]]) -> "ChatHistory":
+        chat_messages = [
+            ChatMessage(role=str(h["role"]), content=str(h["value"]))
```

Comment:
list_of_dicts is typed as `list_of_dicts: list[Dict[str, str]]`, so these extra `str(...)` calls should be unnecessary, unless there is something hidden I don't see

Reply:
You're right, removed.